data_steward/tools/copy_dataset_to_output_prod.py

```diff
@@ -0,0 +1,204 @@
+and runs the recreate_person tool.
+"""
+
+# Python imports
+import logging
+import argparse
+import re
+
+# Project imports
+from utils import bq, auth, pipeline_logging
+from tools.recreate_person import update_person
+
+SCOPES = [
+    'https://www.googleapis.com/auth/bigquery',
+    'https://www.googleapis.com/auth/devstorage.read_write',
+    'https://www.googleapis.com/auth/cloud-platform'
+]
+LOGGER = logging.getLogger(__name__)
+
+TIER_LIST = ['controlled', 'registered']
+DEID_STAGE_LIST = ['deid', 'base', 'clean']
+
+
+def validate_tier_param(tier):
+    """
+    helper function to validate the tier parameter passed is either 'controlled' or 'registered'
+
+    :param tier: tier parameter passed through from either a list or command line argument
+    :return: nothing, breaks if not valid
+    """
+    if tier.lower() not in TIER_LIST:
+        msg = f"Parameter ERROR: {tier} is an incorrect input for the tier parameter, accepted: controlled or " \
+              f"registered"
+        LOGGER.error(msg)
+        raise argparse.ArgumentTypeError(msg)
+
+
+def validate_deid_stage_param(deid_stage):
+    """
+    helper function to validate the deid_stage parameter passed is correct, must be 'deid', 'base' or 'clean'
+
+    :param deid_stage: deid_stage parameter passed through from either a list or command line argument
+    :return: nothing, breaks if not valid
+    """
+    if deid_stage not in DEID_STAGE_LIST:
+        msg = f"Parameter ERROR: {deid_stage} is an incorrect input for the deid_stage parameter, accepted: deid, " \
+              f"base, clean"
+        LOGGER.error(msg)
+        raise argparse.ArgumentTypeError(msg)
+
+
+def validate_release_tag_param(arg_value):
+    """
+    User defined helper function to validate that the release_tag parameter follows the correct naming convention
+
+    :param arg_value: release tag parameter passed through either the command line arguments
+    :return: arg_value
+    """
+
+    release_tag_regex = re.compile(r'[0-9]{4}Q[0-9]R[0-9]')
+    if not re.match(release_tag_regex, arg_value):
+        msg = f"Parameter ERROR {arg_value} is in an incorrect format, accepted: YYYYQ#R#"
+        LOGGER.error(msg)
+        raise argparse.ArgumentTypeError(msg)
+    return arg_value
+
+
+def validate_create_tier_args(tier, stage, tag):
+    """
+    User defined helper function to validate that the tier, deid_stage, release_tag parameter
+     follows the correct naming convention
+    :param tier: tier parameter passed through from either a list or command line argument
+    :param stage: deid_stage parameter passed through from either a list or command line argument
+    :param tag: release tag parameter passed through either the command line arguments
+    :return: None, breaks if not valid params are passed
+    """
+    validate_tier_param(tier)
+    validate_deid_stage_param(stage)
+    validate_release_tag_param(tag)
```

Comment:
This may not be necessary, argparse can validate choices and the string when the args are passed.